### PR TITLE
[ENG-6253] Add tests for GitLab imp

### PR DESCRIPTION
**Pull request by Vlad0n20**

### CODE REVIEW

File: addon_imps/tests/storage/test_gitlab.py [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+    async def test_get_item_info_file_not_found(self):
+        self._patch_get({}, status=HTTPStatus.NOT_FOUND)
+
+        with self.assertRaises(ItemNotFound):
+            self.imp.get_item_info = AsyncMock(
+                side_effect=ItemNotFound("item not found"),
+                spec_set=self.imp.get_item_info,
+            )
+            await self.imp.get_item_info("1:missing_file.md")
+        self.imp.get_item_info.assert_awaited_once_with("1:missing_file.md")
```
Comment (opaduchak) [inline addon_imps/tests/storage/test_gitlab.py:156]: this doesn't test imp implementation's behaviour

File: addon_imps/tests/storage/test_gitlab.py [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+    async def test_get_item_info_file(self):
+        repo_mock = {"default_branch": "main"}
+        file_mock = {
+            "file_name": "README.md",
+            "path": "README.md",
+        }
+
+        self._patch_get(repo_mock)
+        self.network.GET.return_value.__aenter__.return_value.headers = {}
+        self._patch_get(file_mock)
```
Comment (opaduchak) [inline addon_imps/tests/storage/test_gitlab.py:103]: subsequent patches would just override each other

File: addon_imps/tests/storage/test_gitlab.py [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+        self.imp.get_item_info = AsyncMock(
+            spec_set=self.imp.get_item_info,
+            return_value=ItemResult(
+                item_id="1:README.md", item_name="README.md", item_type=ItemType.FILE
+            ),
+        )
+        result = await self.imp.get_item_info("1:README.md")
```
Comment (opaduchak) [inline addon_imps/tests/storage/test_gitlab.py:110]: this also doesn't test imp's behaviour

**Review by opaduchak — CHANGES_REQUESTED** [review]

File: addon_imps/tests/storage/test_gitlab.py [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+        self._assert_get("projects/1")
+
+    async def test_get_item_info_file(self):
+        file_mock = {
+            "file_name": "README.md",
+            "path": "README.md",
+        }
+
+        self._patch_get(file_mock)
+        self.network.GET.return_value.__aenter__.return_value.headers = {}
```
Comment (opaduchak) [inline addon_imps/tests/storage/test_gitlab.py:101]: I'd suggest to implement this in `patch_get`?
But moreover, is this important to specify empty response headers here?

File: addon_imps/tests/storage/test_gitlab.py [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+            },
+        ]
+        self._patch_get(mock_response)
+
+        result = await self.imp.list_root_items()
+        expected_items = [
+            ItemResult(item_id="repo1:", item_name="repo1", item_type=ItemType.FOLDER),
+            ItemResult(item_id="repo2:", item_name="repo2", item_type=ItemType.FOLDER),
+        ]
+        expected_result = ItemSampleResult(items=expected_items)
```
Comment (opaduchak) [inline addon_imps/tests/storage/test_gitlab.py:66]: may you also add tests for pagination?

**Review by opaduchak — CHANGES_REQUESTED** [review]

File: addon_imps/tests/storage/test_gitlab.py (hunk 4 above)
Comment (Vlad0n20) [reply addon_imps/tests/storage/test_gitlab.py:101]: Actually, it is not necessary

File: addon_imps/tests/storage/test_gitlab.py (hunk 5 above)
Comment (Vlad0n20) [reply addon_imps/tests/storage/test_gitlab.py:66]: Sure

File: addon_imps/tests/storage/test_gitlab.py [hunk 6]
```diff
@@ -0,0 +1,211 @@
...
+        self._patch_get(folder_mock_page_2, headers={})
+        result_page_2 = await self.imp.list_child_items("1:", page_cursor="page=2")
+        expected_items_page_2 = [
+            ItemResult(
+                item_id="1:LICENSE", item_name="LICENSE", item_type=ItemType.FILE
+            ),
+        ]
+
+        self.assertEqual(result_page_2.items, expected_items_page_2)
+        self.assertIsNone(result_page_2.next_sample_cursor)
```
Comment (opaduchak) [inline addon_imps/tests/storage/test_gitlab.py:211]: Please split this into separate tests to be sure that we have clean environment for each case

File: addon_imps/tests/storage/test_gitlab.py [hunk 7]
```diff
@@ -0,0 +1,211 @@
...
+            {"id": "3", "name": "repo3", "path_with_namespace": "repo3"},
+        ]
+        self._patch_get(mock_response_page_2, headers={})
+        result_page_2 = await self.imp.list_root_items(page_cursor="page=2")
+        expected_items_page_2 = [
+            ItemResult(item_id="repo3:", item_name="repo3", item_type=ItemType.FOLDER),
+        ]
+
+        self.assertEqual(result_page_2.items, expected_items_page_2)
+        self.assertIsNone(result_page_2.next_sample_cursor)
```
Comment (opaduchak) [inline addon_imps/tests/storage/test_gitlab.py:176]: Please split this into separate tests to be sure that we have clean environment for each case

**Review by opaduchak — CHANGES_REQUESTED** [review]

**Review by brianjgeiger — APPROVED** [review]

**Comment by brianjgeiger** [comment]
@opaduchak Are all the changes made to your satisfaction? If so, could you approve this PR?

**Review by opaduchak — APPROVED** [review]

**Pull request merged**